src/main/java/io/deephaven/hash/KeyedIntObjectHash.java

```diff
@@ -252,15 +253,15 @@ public synchronized boolean replace(Integer key, V oldValue, V newValue) {
       throw new IllegalArgumentException(
           "key and value are inconsistent:" + key + " and " + intKeyDef.getIntKey(newValue));
     }
-    return internalPut(newValue, KeyedIntObjectHash.REPLACE, oldValue) != null;
+    return Objects.equals(internalPut(newValue, KeyedIntObjectHash.REPLACE, oldValue), oldValue);
```

Comment:
I think this `equals` (in all 5 cases across the 3 files) is unnecessary/overkill. These maps cannot hold `null` values. You could instead do:
```
if (oldValue == null) {
    throw new NullPointerException("oldValue is null, but this map cannot hold null values");
}
```

Reply:
Passing `null` to `internalPut` does that. That doesn't mean passing `null` to `replace` should do that.